### PR TITLE
Implement suggestions for unknown features in workspace

### PR DESCRIPTION
**Pull request by In-line**

### CODE REVIEW

File: src/cargo/core/workspace.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@ use std::slice;
 
 use anyhow::{bail, Context as _};
 use glob::glob;
+use itertools::*;
```
Comment (ehuss) [inline src/cargo/core/workspace.rs:10]: Can you change this glob import to import only the traits needed?

**Comment by ehuss** [comment]
Thanks for the PR!

I'm having trouble getting some of the suggestions to work, and I was wondering if there is something not working as expected.  I can't get anything with `/` syntax to show a warning, but it looks like the code is intending for it to work. Some examples from Cargo's own workspace:

```
> cargo build --features cargo/deny-warning
error: none of the selected packages contains these features: cargo/deny-warning

> cargo build --features openssl/vendoredd
error: failed to select a version for `openssl`.
    ... required by package `cargo v0.54.0 (/Users/eric/Proj/rust/cargo2)`
versions that meet the requirements `=0.10.33` are: 0.10.33

the package `cargo` depends on `openssl`, with features: `vendoredd` but `openssl` does not have these features.


failed to select a version for `openssl` which could resolve this conflict
```

I am able to get features without `/` to work (although `lev_distance` isn't always the greatest for catching typos).

**Review by ehuss — COMMENTED** [review]
I'm a little leery of adding itertools as a dependency, but it doesn't seem too heavy, and I can't deny that it has some useful helpers.

File: tests/testsuite/package_features.rs [hunk 2]
```diff
@@ -67,13 +67,53 @@ fn virtual_no_default_features() {
...
+#[cargo_test]
+fn virtual_typo_member_feature() {
+    project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "a"
+            version = "0.1.0"
+            resolver = "2"
```
Comment (In-line) [inline tests/testsuite/package_features.rs:103]: See #9437

**Comment by ehuss** [comment]
Thanks!  Sorry about the delay.

@bors r+

**Pull request merged**